spinalcordtoolbox/image.py

```diff
@@ -501,9 +509,17 @@ def save(self, path=None, dtype=None, verbose=1, mutable=False):
         hdr = self.hdr.copy() if self.hdr else None
         if hdr:
             hdr.set_data_shape(data.shape)
-            # Update dtype if provided (but not if based on SCT-specific values: 'minimize')
-            if (dtype is not None) and (dtype not in ['minimize', 'minimize_int']):
-                hdr.set_data_dtype(dtype)
+            # Update dtype to match that of the data array
+            dtype_data = data.dtype
+            dtype_header = hdr.get_data_dtype()
+            if dtype_data != dtype_header:
+                logger.warning(f"Image header specifies datatype '{dtype_header}', but array is of type"
+                               f"type '{dtype_data}'. Header metadata will be overwritten to use '{dtype_data}'.")
```

Comment:
```suggestion
                logger.warning(f"Image header specifies datatype '{dtype_header}', but array is of type"
                               f"'{dtype_data}'. Header metadata will be overwritten to use '{dtype_data}'.")
```

Reply:
The second, copy-pasted typo was removed by the refactoring of the two separate checks.